autosubmit_api/experiment/common_requests.py

```diff
@@ -244,12 +245,13 @@ def _is_exp_running(expid: str, time_condition=300) -> Tuple[bool, str, bool, in
         # print("Experiment {0} version {1} \nLook {2} \nLook {3}".format(
         #     expid, current_version, pathlog_first, pathlog_second))
         # print(pathlog)
-        reading = os.popen(
-            'ls -t ' + pathlog_first + ' | grep "_run.log"').read() if (os.path.exists(pathlog_first)) else ""
+        dir_files = []
+        if os.path.exists(pathlog_first):
+            dir_files = get_files_from_dir_with_pattern(pathlog_first, "_run.log")
 
         #print("Length {0}".format(len(reading)))
-        if (reading) and len(reading) > 0:
-            log_file_name = reading.split()[0]
+        if (dir_files) and len(dir_files) > 0:
```

Comment:
```suggestion
        if dir_files:
```

That should check both that `dir_files` is not `None` and that it is not empty,

```py
>>> dir_files = None
>>> assert dir_files
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> dir_files = []
>>> assert dir_files
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> dir_files = [1, 2]
>>> assert dir_files
```

Reply:
You are right, the second condition is redundant